virttest/env_process.py

```diff
@@ -1164,26 +1162,6 @@ def preprocess(test, params, env):
                 vm.destroy()
                 del env[key]
 
-    if (params.get("auto_cpu_model") == "yes" and
-            vm_type == "qemu"):
-        if not env.get("cpu_model"):
-            env["cpu_model"] = cpu.get_qemu_best_cpu_model(params)
-        params["cpu_model"] = env.get("cpu_model")
-
-    if vm_type == "qemu":
-        qemu_path = utils_misc.get_qemu_binary(params)
-        if (utils_qemu.has_device_category(qemu_path, "CPU")
-                and params.get("cpu_driver") is None):
-            cpu_model = params.get("cpu_model")
-            if cpu_model:
```

Comment:
Would you please elaborate on why this check can be removed for the new approach?

Reply:
Because cpu_model is always assigned a value via L1626~L1642 of qemu_vm.py.

Reply:
What if the configuration would be like this?
```
auto_cpu_model = no
cpu_model =
default_cpu_model =
```

Reply:
That's like an unreasonable misconfiguration to me. Do we need to handle all the user configuration error? It will fail all the time.

Reply:
Well, qemu allows users to launch VMs without setting `-cpu`, so from my point of view it's not a misconfiguration.
```
$ qemu-kvm -m 512 -monitor stdio
QEMU 5.2.0 monitor - type 'help' for more information
(qemu) 
```

Reply:
Added the check back, I guess we better not change the behavior. Thanks.

Reply:
> Well, qemu allows users to launch VMs without setting `-cpu`, so from my point of view it's not a misconfiguration.
> 
> ```
> $ qemu-kvm -m 512 -monitor stdio
> QEMU 5.2.0 monitor - type 'help' for more information
> (qemu) 
> ```

Make sense. I missed the scenario. Pls help review again.